src/components/baseContainer/BaseContainer.tsx

```diff
@@ -1,9 +1,27 @@
-import {View} from 'react-native';
+import {SafeAreaView} from 'react-native';
+import {SafeAreaProvider} from 'react-native-safe-area-context';
+
+import CustomStatusBar from '@components/customStatusBar';
 import useTheme from '@hooks/useTheme';
 
-const BaseContainer = ({children}: {children: React.ReactNode}) => {
-  const {layout, backgrounds} = useTheme();
-  return <View style={[layout.flex_1, backgrounds.white]}>{children}</View>;
+const BaseContainer = ({
+  children,
+  showStatusBar = true,
+  bgColor = 'white',
+}: {
+  children: React.ReactNode;
+  showStatusBar?: boolean;
+  bgColor?: string;
+}) => {
+  const {layout} = useTheme();
+  return (
+    <SafeAreaProvider>
```

Comment:
purpose of using this in base container file and why status bar is not handles in main App.tsx file

Reply:
So that I can manage where to status bar and where not. If I am buliding any screen where I don't need status bar, I can pass showStatusBar = {false}.

Reply:
but status bar should be same in whole app